include/master_element/Quad42DCVFEM.h

```diff
@@ -0,0 +1,194 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+
+#ifndef Quad42DCVFEM_h   
+#define Quad42DCVFEM_h   
+
+#include <master_element/MasterElement.h>
+
+#include <AlgTraits.h>
+
+// NGP-based includes
+#include "SimdInterface.h"
+#include "KokkosInterface.h"
+
+#include <vector>
+#include <cstdlib>
+#include <stdexcept>
+#include <string>
+#include <array>
+
+#ifdef __INTEL_COMPILER
+#define POINTER_RESTRICT restrict
+#else
+#define POINTER_RESTRICT __restrict__
+#endif
+
+namespace sierra{
+namespace nalu{
+
+// 2D Quad 4 subcontrol volume
+class Quad42DSCV : public MasterElement
+{
+public:
+  Quad42DSCV();
+  virtual ~Quad42DSCV();
+
+  const int * ipNodeMap(int ordinal = 0);
+
+  void determinant(
```

Comment:
@overfelt With the merge of Alan's Pyramid NGP modifications, the method signature has changed to pass-by-reference instead of pass-by-value.